File: src/030/p030.hs
import Data.Digits

digitsFifthPower :: Integer -> Integer
digitsFifthPower n =
  sum $ map (\x -> x ^ 5) $ digits 10 n

solve :: Integer
solve =
  sum $ filter (\x -> digitsFifthPower x == x) $ takeWhile (< 200000) [1..]

main :: IO ()
main =
  print $ solve

File: src/030/p030.exs
 defmodule Problem030 do
  defp digits_fifth_power(x) do
    x
    |> Integer.digits()
    |> Enum.map(&(trunc(:math.pow(&1, 5))))
    |> Enum.sum()
  end

  def solve do
    Stream.iterate(2, &(&1+1))
    |> Stream.take_while(&(&1 < 200_000))
    |> Stream.filter(&(&1 == digits_fifth_power(&1)))
    |> Enum.sum()
  end
end

IO.puts Problem030.solve
